### PR TITLE
Bump golangci-lint to 1.43.0

### PR DESCRIPTION
**Pull request by stmcginnis**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

This version of golangci-lint pulls in releases of its linters that
address issues when running under go 1.17. With these newer linters,
there are also better linting checks.

This bumps the golangci-lint version and addresses the one additional
issue this newer version raised.

**Special notes for your reviewer**:

Can be validated by running `make lint-full`

**TODOs**:

- [X] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: controllers/helpers_test.go [hunk 1]
```diff
@@ -14,6 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+//nolint:gosec
```
Comment (mboersma) [inline controllers/helpers_test.go:17]: Could you instead put this skip statement at line 405? Then the real code in the top half of this file will still be checked by `gosec`.

**Review by mboersma — CHANGES_REQUESTED** [review]

File: controllers/helpers_test.go (hunk 1 above)
Comment (stmcginnis) [reply controllers/helpers_test.go:17]: Sure thing - I was thinking this was all test code based on the name, so skipping it all. But in general it is probably better to be more specific anyway. Thanks!

**Review by mboersma — APPROVED** [review]
/lgtm

**Review by CecileRobertMichon — COMMENTED** [review]
/approve

**Pull request merged**